pandas/io/pytables.py

```diff
@@ -1691,29 +1691,37 @@ class IndexCol:
     is_data_indexable = True
     _info_fields = ["freq", "tz", "index_name"]
 
+    name: str
```

Comment:
I think had this discussion on another PR can't remember where we landed, but do these declarations need to be in the class scope? Not sure every reader would know why these are there and may in fact infer that they should be class variables.

Reply:
This topic has definitely come up and the conclusion may have just gone over my head.  When I see this, I read it as "`self.name` is going to be defined in `__init__`, and is always going to be a `str`.  This is helpful to me, the reader, since this class seems to set its attributes all over the place"

Reply:
Gotcha. Actually reading through PEP 526 this looks to be correct, so nice find

https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations

@simonjayhawkins